src/crypto/internal/boring/openssl_port_rsa.c

```diff
@@ -43,15 +45,15 @@ int _goboringcrypto_RSA_digest_and_sign_pss_mgf1(GO_RSA *rsa, unsigned int *out_
 
 	if (_goboringcrypto_EVP_PKEY_sign_init(ctx) <= 0)
 		goto err;
+	if (_goboringcrypto_EVP_PKEY_CTX_set_signature_md(ctx, md) <= 0)
```

Comment:
This is moved earlier because openssl 3.0 defaults to sha1 in EVP_PKEY_CTRL_RSA_PADDING if unset and produces an error